src/passes/analysis/verilog.cpp

```diff
@@ -461,16 +461,12 @@ Passes::Verilog::compilePorts(RecordType *record_type) {
       ASSERT(false, "Not implemented for type = " + toString(type));
     }
     std::unique_ptr<vAST::Port> port = std::make_unique<vAST::Port>(
-            process_decl(std::move(name), type), verilog_direction, vAST::WIRE);
+            process_decl(std::move(name), type), verilog_direction,
+            vAST::WIRE);
     if (this->verilator_debug) {
-      // FIXME: Hack to get comment into port decl, we need to add support
-      // attaching comments to expressions
-      std::string port_str = port->toString();
-      port_str += "/*verilator public*/";
-      ports.push_back(std::make_unique<vAST::StringPort>(port_str));
```

Comment:
we should get rid of the `StringPort` class/node if not done already.

Reply:
Unfortunately StringPort is still used by the primitive code that uses a verilog string as the body and a list of strings for the interface ports (i.e. `verilog_json["interface"]` inside `compileStringBodyModule`), so we'll need to update that code path (mainly move all the primitives to use verilogAST) for that to be removed.